docs: update releasing and support policy (#2527)
AmadeusITGroup/otter: SECURITY.md

## Release version support

The "support" of a released version includes the following aspects:

> - Handle bug fixing when requested via the [Github Issue page](https://github.com/AmadeusITGroup/otter/issues).

Reviewer: ```suggestion
- Handling bug fixes when requested via the [Github Issue page](https://github.com/AmadeusITGroup/otter/issues).
```

> - Review (and handle merging) of bugfix pull requests contribution.

Reviewer: ```suggestion
- Reviewing contributions to bugfix pull requests (and handling their merging).
```

> - Cascade bugfixes when contributed into inferior released version.

Reviewer: ```suggestion
- Cascading bugfixes when they are implemented in an inferior released version.
```

> - Handle bugfix cherry-picking on specific version demand.

Reviewer: ```suggestion
- Handling bugfix cherry-picking on a specific version demand.
```

> [!WARNING]

> > Only **bugfix** are accepted on supported versions. New features are allowed exclusively on the `main` (latest *prerelease* version) and the `next` branches.

Reviewer: ```suggestion
> Only **bugfixes** are accepted on supported versions. New features are allowed exclusively on the `main` (latest *prerelease* version) and the `next` branches.
```

> > Only **bugfixes** are accepted on supported versions. New features are allowed exclusively on the `main` (latest *prerelease* version) and the `next` branches.

Reviewer 2: **Bugfixes** are only accepted on supported versions ?

> Note that **breaking changes** should always target a `next` branch.

The team will support the latest minor version of major versions **N (current)**, **N-1** and **N-2**.\

> All the minors versions of the major **N (current)** are also supported as long as a new major version is not created.

Reviewer: ```suggestion
All the minor versions of the major **N (current)** are also supported until a new major version is created.
```

> The major becoming **N-3** will also be flagged as unsupported.

Reviewer: ```suggestion
The major version becoming **N-3** will also be flagged as unsupported.
```

## Release version support

The "support" of a released version includes the following aspects:

- Handle bug fixing when requested via the [Github Issue page](https://github.com/AmadeusITGroup/otter/issues).
- Review (and handle merging) of bugfix pull requests contribution.
- Cascade bugfixes when contributed into inferior released version.
- Handle bugfix cherry-picking on specific version demand.

> [!WARNING]

> > Only **bugfix** are accepted on supported versions. A new feature are allowed exclusively on the `main` branch (latest *alpha* version).

Reviewer 3: ```suggestion
> Only **bugfix** are accepted on supported versions. A new feature are allowed exclusively on the `main` branch (latest *prerelease* version) and the `next` branch.
```

> > Only **bugfix** are accepted on supported versions. A new feature are allowed exclusively on the `main` (latest *prerelease* version) and the `next` branches.

Reviewer 4: ```suggestion
> Only **bugfix** are accepted on supported versions. New features are allowed exclusively on the `main` (latest *prerelease* version) and the `next` branches. Note that breaking changes should always target a `next` branch.
```

The team will support the latest minor version of major versions **N (current)**, **N-1** and **N-2**.\
All the minors versions of the major **N (current)** are also supported as long as a new major version is not created.

When a new major version **N** is released, all the minors version of the major **N-1**, except the latest one, will be considered as non-supported.\

> The major becoming **N-3** will be also flagged as non-supported.

Reviewer 4: ```suggestion
The major becoming **N-3** will also be flagged as non-supported.
```

> A major version will be supported as long as it relies on a [non deprecated version of Angular](https://angular.io/guide/releases#support-policy-and-schedule) and included in the previously listed major versions.

Reviewer 4: ```suggestion
A major version will be supported as long as it relies on a [non-deprecated version of Angular](https://angular.io/guide/releases#support-policy-and-schedule) and is included in the previously listed major versions.
```
